portfolio/src/main/webapp/script.js

```diff
@@ -56,6 +56,17 @@ function interests() {
 
 /** Creates a map and adds it to the page. */
 function loadMap() {
+    var contentString = '<div id="map-content">'+
+      '<div id="siteNotice">'+
+      '</div>'+
+      '<h1 id="firstHeading" class="firstHeading">'+document.getElementById("locations").value+'</h1>'+
+      '<div id="bodyContent">'+
+      '<p> This place is awesome!</p>'+
+      '<p>Unrelated Attribution: <a href="https://rustacean.net/assets/rustacean-flat-happy.svg">'+
+      'Unrelated Picture</a> '+
+      '</p>'+
+      '</div>'+
+      '</div>';
```

Comment:
What you're doing here is creating an html element with js code. This is not easy to read or debug. It's basically string snippets stitched together. I'm actually not sure what is the right way to do this. Maybe the right way is this. Maybe we should use document.createElement and build DOM objects programmatically. Another approach would be to add this div into your main html file but keep it hidden (`x.style.display === "none"`). Then copy it into your map info window as it loads.

Reply:
ah ok-I've used the second approach before I think. I'll opt for that first and see how it looks!

Reply:
hmm actually on second thought, I'm not sure how to approach this, because this isn't just a normal set of HTML added to the DOM directly but it uses the maps API to be displayed

Reply:
any suggestions on how to address that difference?

Reply:
For the benefit of time, let's keep this as is and move one. We can revisit this problem in the capstone map view.